### PR TITLE
Ported template to SCSS

### PR DESCRIPTION
**Pull request by lukebrooker**
Great work with all of this. I am using grunt, font-smith and this to generate icons as soon as I drop an svg into a folder. I need this template so I can use the project in a team environment and it is included when dependencies are pulled from npm.

With this implementation, I tried to do an exact replica of your current templates.

If you are interested, I did a slightly different version in my local copy which I think is a little more flexible:
https://gist.github.com/lukebrooker/6694910

Not sure if it is possible with LESS though.

### CODE REVIEW

**Comment by twolfson** [comment]
Awesome! At first glance, everything looks good. However, I want to spend more time in the details. I should take a look by the end of the week, worst case Sunday.

...and yes, your gist is not possible in LESS; it does not support arrays, let alone nesting objects in arrays.

**Comment by lukebrooker** [comment]
Yeah, no worries. I just realised I uncommented the test and didn't change it, so it still has the sprite stuff in there, sorry :(

File: lib/templates/scss.mustache.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+@mixin icon($char) {
+  @include iconFontFamily($char)
+  @include iconFont()
+
+  &:before {
+    @include iconContent($char)
+  }
+}
+
```
Comment (lukebrooker) [inline lib/templates/scss.mustache.scss:33]: I left the semicolons out here too :(

**Comment by twolfson** [comment]
Sorry, still haven't gotten time to review the PR.

You can push new changes to a PR branch and GitHub will pick up the changes.

File: test/json2fontcss_content.js [hunk 2]
```diff
@@ -153,35 +153,35 @@ module.exports = {
...
+    var tmp = new Tempfile();
+    tmp.writeFileSync(scssStr);
+    exec('sass --scss ' + tmp.path, function (err, css, stderr) {
+      assert.strictEqual(stderr, '');
+      assert.strictEqual(err, null);
+      assert.notEqual(css, '');
+      // console.log('SCSS', css);
+      done(err);
+    });
   }
```
Comment (twolfson) [inline test/json2fontcss_content.js:186]: In addition to uncommenting these, you needed to add the appropriate skeletons to `json2fontcss_outline`. This is a nuance of doubleshot though =P

Comment (twolfson) [inline test/json2fontcss_content.js:186]: Oh noes, I added the outlines and tests are failing. I am taking an interlude to add Travis CI but I will come back to them.

Comment (twolfson) [inline test/json2fontcss_content.js:186]: Okay... done with Travis (released in `0.1.2`).

File: test/expected_files/scss.scss [hunk 3]
```diff
@@ -1,49 +1,47 @@
...
+$icon2-value: "\e0001";
+$icon2: "my-icon-set" "\e0001";
+$icon3-font-family: "my-icon-set";
+$icon3-value: "\e0002";
+$icon3: "my-icon-set" "\e0002";
 
-  @mixin sprite-width($sprite) {
-    width: nth($sprite, 5);
-  }
+@mixin iconFontFamily($char) {
```
Comment (twolfson) [inline test/expected_files/scss.scss:11]: I believe the standard to SASS/SCSS is `hyphen-case`.

File: test/expected_files/scss.scss [hunk 4]
```diff
@@ -1,49 +1,47 @@
...
+  font-family: nth($char,1);
```
Comment (twolfson) [inline test/expected_files/scss.scss:12]: Putting a space after your comma is nice for separating symbols (improves readability).

```
nth($char, 1);
```

File: test/expected_files/scss.scss [hunk 5]
```diff
@@ -1,49 +1,47 @@
...
-  }
+@mixin iconContent($char) {
+  content: nth($char,2);
+}
 
-  @mixin sprite-image($sprite) {
-    background-image: url(nth($sprite, 7));
-  }
+@mixin icon($char) {
+  @include iconFontFamily($char)
```
Comment (twolfson) [inline test/expected_files/scss.scss:30]: Semicolons are required by SCSS on these lines.

File: test/json2fontcss_content.js [hunk 6]
```diff
@@ -153,35 +153,35 @@ module.exports = {
...
+    scssStr += '\n' + [
+      '.feature {',
+      '  height: $sprite1-height;',
+      '  @include sprite-width($sprite2);',
+      '  @include sprite-image($sprite3);',
+      '}',
+      '',
+      '.feature2 {',
+      '  @include sprite($sprite2);',
+      '}'
```
Comment (twolfson) [inline test/json2fontcss_content.js:173]: This section should have been based off of the other sections. It was originally copy/pasted from `json2css` from the `spritesmith` project.

**Comment by twolfson** [comment]
Alright, I made the adjustments I commented on and released in `0.2.0`. I will be updating `grunt-fontsmith` in a bit.

**Comment by twolfson** [comment]
Okay, `grunt-fontsmith@0.3.0` contains the latest `json2fontcss`. Thanks again for the work on the PR! =)

**Comment by lukebrooker** [comment]
Thanks so much for this. Sorry it was not the best PR, I will take more care next time.

**Comment by twolfson** [comment]
No worries. It was much better than most =)

**Pull request merged**